idaes/core/util/model_diagnostics.py

```diff
@@ -3653,6 +3653,72 @@ def check_parallel_jacobian(model, tolerance: float = 1e-4, direction: str = "ro
 
     jac, nlp = get_jacobian(model, scaled=False)
 
+    # Get vectors that we will check, and the Pyomo components
+    # they correspond to.
+    if direction == "row":
+        components = nlp.get_pyomo_constraints()
+        mat = jac.tocsr()
+
+    elif direction == "column":
+        components = nlp.get_pyomo_variables()
+        mat = jac.transpose().tocsr()
+
+    norms = [norm(mat[i, :], ord="fro") for i in range(len(components))]
+
+    # Take product of all rows/columns with all rows/columns by taking outer
+    # product of matrix with itself
+    outer = mat @ mat.transpose()
+    # Get rid of duplicate values by only taking upper triangular part of
+    # resulting matrix
+    upper_tri = triu(outer)
+    # List to store pairs of parallel components
+    parallel = []
+
+    for row, col, val in zip(*find(upper_tri), strict=True):
```

Comment:
When I try to test this, I get an error due to this keyword argument passed to `zip`. I assume `strict` should be passed to `triu`, but I'm not sure exactly what else is going on here.

Reply:
Sorry about that. No, `strict` was an argument for `zip`. It throws an error if the iterators passed to `zip` are not the same length. However, it appears it was only added in Python 3.10, so I can't use it.

For reference `find` returns a tuple of three arrays, one containing row indices, one containing column indices, and one containing matrix values for all the nonzero entries of `upper_tri`. `*` unpacks the tuple into arguments for `zip`. It's a neater way of writing `zip(upper_tri[0], upper_tri[1], upper_tri[2])`.